source/hooks/useOnlyOnce/index.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * A useEffect wrapper that will run a function once and only once, on the first component render
+ */
+ import { useEffect } from 'react';
+
+const registry = {};
+
+const useOnlyOnce = (callback, name) => {
+    if (!name) {
+        throw new Error('useOnlyOnce requires a name for the callback');
+    }
+
+    if (!registry[name]) {
+        useEffect(callback, []);
```

Comment:
This breaks a rule of hooks. https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level